sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/translation/impl/ScalarFnImpl.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.sql.zetasql.translation.impl;
```

Comment:
Why is this in `zetasql`? How will it work for the calcite dialect?

Reply:
I think the PR description mentions the reason: `The Method will be passed to createUdfOperator in a future PR.`

Reply:
How will that work for the calcite dialect?

Reply:
There's nothing specific to ZetaSQL in this class, I can move it somewhere else.

Reply:
`createUdfOperator` is actually in ZetaSQL only now. But I agree we can move this code to the sql module and later might bring `createUdfOperator` to that module as well.

Reply:
Great. My concern here is that there already exists code to do this for Calcite SQL, if so we should just reuse that. It looks like Calcite SQL support just wasn't part of the prototype?

Reply:
> My concern here is that there already exists code to do this for Calcite SQL, if so we should just reuse that.

We do something similar in Calcite SQL, but not exactly because the interfaces are different.

> It looks like Calcite SQL support just wasn't part of the prototype?

Yeah, how (if at all) we want to integrate the new APIs into Calcite SQL is an open question.

Reply:
What I hope happens it that the UDFs are loaded into the schema in BeamSqlEnvBuilder, then both Calcite and ZetaSQL dialects pull them from there.

Reply:
> What I hope happens it that the UDFs are loaded into the schema in BeamSqlEnvBuilder, then both Calcite and ZetaSQL dialects pull them from there.

I will try doing that for the next PR in this series.